### PR TITLE
feat(s3): Create specified bucket when it doesn't exist

### PR DESCRIPTION
**Pull request by lwander**
If you prefer `ensureBucketExists` does not get called during init that's OK. I filled out that method primarily for Halyard's consumption.

### CODE REVIEW

File: front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java [hunk 1]
```diff
@@ -50,22 +40,35 @@
...
     this.bucket = bucket;
     this.rootFolder = rootFolder;
     this.readOnlyMode = readOnlyMode;
+    this.region = region;
   }
 
   @Override
   public void ensureBucketExists() {
-    // do nothing
+    List<Bucket> buckets = amazonS3.listBuckets();
```
Comment (robzienert) [inline front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java]: [headBucket](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#headBucket-com.amazonaws.services.s3.model.HeadBucketRequest-) is probably a better fit here than iterating all existing buckets.

Comment (lwander) [inline front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java]: Oh neat, I was looking for something like that. Thanks

Comment (lwander) [inline front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java]: Updated

**Comment by robzienert** [comment]
👍

**Comment by tomaslin** [comment]
Please use naming conventions http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch

**Comment by lwander** [comment]
Naming conventionalized

**Comment by robzienert** [comment]
Wanna rebase & we'll merge this guy?

**Comment by lwander** [comment]
We can get the same effect with `Update branch` + `Squash and merge`!

**Pull request merged**